app/api/groups/get_group_progress.go

```diff
@@ -0,0 +1,175 @@
+				groups_ancestors.idGroupChild != groups_ancestors.idGroupAncestor AND
+				groups_ancestors.idGroupAncestor IN (?)`, ancestorGroupIDs). // preliminary filter (makes the query faster)
+		Joins(`JOIN groups ON groups.ID = groups_ancestors.idGroupChild AND groups.sType IN ('UserSelf', 'Team')`).
+		Joins(`
+			JOIN groups_attempts AS attempt_with_best_score
+			ON attempt_with_best_score.ID = (
+				SELECT ID FROM groups_attempts
+				WHERE idGroup = main_attempts.idGroup AND idItem = main_attempts.idItem
+				ORDER BY idGroup, idItem, iMinusScore, sBestAnswerDate LIMIT 1
+			)`).
+		Where("main_attempts.idItem IN (?)", itemIDs). // preliminary filter (makes the query faster)
+		Group("main_attempts.idGroup, main_attempts.idItem")
+
+	endMembers := srv.Store.GroupAncestors().
+		Select("groups_ancestors.idGroupAncestor AS idGroup, end_member.ID AS idEndMember").
+		Joins(`
+			JOIN groups AS end_member
+			ON end_member.ID = groups_ancestors.idGroupChild AND end_member.sType IN ('UserSelf', 'Team')`).
+		// The team users filtering starts here (is not well specified yet)
+		Joins(`
+			LEFT JOIN groups_groups AS team_link
+			ON
+				end_member.ID IS NOT NULL AND
+				end_member.sType != 'Team' AND
+				team_link.idGroupChild = end_member.ID AND
+				team_link.sType IN('invitationAccepted','requestAccepted','direct')`).
+		Joins(`
+			LEFT JOIN groups_ancestors AS team_ancestor_check
+			ON team_ancestor_check.idGroupChild = team_link.idGroupParent AND
+				team_ancestor_check.idGroupAncestor = groups_ancestors.idGroupAncestor`).
+		Joins("LEFT JOIN groups AS team ON team.ID = team_ancestor_check.idGroupChild AND team.sType = 'Team'").
+		Joins(`
+			LEFT JOIN (SELECT 1 as bHasOtherParents) AS other_parents
+			ON team.ID IS NOT NULL AND (
+				SELECT 1
+				FROM groups_groups
+				WHERE idGroupChild = end_member.ID AND
+					idGroupParent != team.ID AND
+					idGroupParent IN (SELECT idGroupChild FROM groups_ancestors AS ga WHERE ga.idGroupAncestor = groups_ancestors.idGroupAncestor)
+				LIMIT 1
+			) = 1`).
+		// The team users filtering ends here
+		Where("groups_ancestors.idGroupAncestor IN (?)", ancestorGroupIDs). // preliminary filter (makes the query faster)
+		Where("groups_ancestors.idGroupAncestor != groups_ancestors.idGroupChild").
+		Group("groups_ancestors.idGroupAncestor, end_member.ID").
+		// The team users filtering (is not well specified yet)
+		Having("MIN(end_member.sType = 'Team' OR (team.ID IS NULL OR (team.ID IS NOT NULL AND other_parents.bHasOtherParents)))")
+
+	// We want to keep groups that don't have end members
+	groupsEndMembers := srv.Store.Groups().
+		Select("groups.ID AS idGroup, end_member.idEndMember AS idEndMember").
+		Joins(`
+			LEFT JOIN ? AS end_member
+			ON end_member.idGroup = groups.ID`, endMembers.SubQuery()).
+		Where("groups.ID IN (?)", ancestorGroupIDs)
+
+	var dbResult []map[string]interface{}
+	service.MustNotBeError(srv.Store.Raw(`
```

Comment:
Detail 🙂. Isn't it possible to write without "raw"? It is easier to read when the parameters are used directly on the same "line" and not at the very end of query.

Reply:
The "Raw" has gone.